crowdfunding/projects/serializers.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework import serializers
+
+from .models import Project, Pledge
+from users.serializers import CustomUserSerializer
+
+class PledgeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Pledge
+        fields = ['id', 'amount', 'comment', 'anonymous', 'project', 'supporter'] #or fields = '__all__'
+        read_only_fields = [id,'supporter']
```

Comment:
your pledge serializer will still show the details for an anonymous pledge, which probably should be avoided.